sbi/utils/plot/plot.py

```diff
@@ -0,0 +1,800 @@
+import numpy as np
+from matplotlib import pyplot as plt
+
+
+def _update(d, u):
+    # https://stackoverflow.com/a/3233356
+    for k, v in six.iteritems(u):
+        dv = d.get(k, {})
+        if not isinstance(dv, collectionsAbc.Mapping):
+            d[k] = v
+        elif isinstance(v, collectionsAbc.Mapping):
+            d[k] = _update(dv, v)
+        else:
+            d[k] = v
+    return d
+
+
+def _format_axis(ax, xhide=True, yhide=True, xlabel="", ylabel="", tickformatter=None):
+    for loc in ["right", "top", "left", "bottom"]:
+        ax.spines[loc].set_visible(False)
+    if xhide:
+        ax.set_xlabel("")
+        ax.xaxis.set_ticks_position("none")
+        ax.xaxis.set_tick_params(labelbottom=False)
+    if yhide:
+        ax.set_ylabel("")
+        ax.yaxis.set_ticks_position("none")
+        ax.yaxis.set_tick_params(labelleft=False)
+    if not xhide:
+        ax.set_xlabel(xlabel)
+        ax.xaxis.set_ticks_position("bottom")
+        ax.xaxis.set_tick_params(labelbottom=True)
+        if tickformatter is not None:
+            ax.xaxis.set_major_formatter(tickformatter)
+        ax.spines["bottom"].set_visible(True)
+    if not yhide:
+        ax.set_ylabel(ylabel)
+        ax.yaxis.set_ticks_position("left")
+        ax.yaxis.set_tick_params(labelleft=True)
+        if tickformatter is not None:
+            ax.yaxis.set_major_formatter(tickformatter)
+        ax.spines["left"].set_visible(True)
+    return ax
+
+
+def get_colors(normalize=True):
+    hex2rgb = lambda h: tuple(int(h[i : i + 2], 16) for i in (0, 2, 4))
+
+    # RGB colors in [0, 255]
+    col = {}
+
+    # parameters and corresponding data
+    col["POSTERIOR1"] = hex2rgb("2E7FB6")
+    col["POSTERIOR2"] = hex2rgb("5DB784")
+    col["POSTERIOR3"] = hex2rgb("EDE249")
+    col["CONSISTENT1"] = hex2rgb("8D62BC")
+    col["CONSISTENT2"] = hex2rgb("AF99EF")
+    col["INCONSISTENT"] = hex2rgb("D73789")
+    col["GT"] = hex2rgb("30C05D")
+
+    # methods
+    col["SNPE"] = hex2rgb("2E7FE8")
+    col["MCMC"] = hex2rgb("FFDF50")
+
+    # rarely used
+    col["PRIOR"] = hex2rgb("2EDEE8")
+
+    # Convert to RGB colors in [0, 1]
+    if normalize:
+        for k, v in col.items():
+            col[k] = tuple([i / 255 for i in v])
+
+    return col
+
+
+def samples_nd(samples, points=[], **kwargs):
+    """Plot samples and points
+
+    See `opts` below for available keyword arguments.
```

Comment:
`opts` should become keyword args to the function. That way, we'll have documentation on the docs. I originally moved them into `opts` for faster development, when I wasn't yet sure what will be useful.

Reply:
You mean, get rid of the `opts` dictionary all together?

Reply:
Yes, I'd make them all keyword arguments instead so that they can be documented

Reply:
okey makes sense